sdk/src/test/scala/com/horizen/account/state/ForgerStakeMsgProcessorTest.scala

```diff
@@ -656,8 +656,173 @@ class ForgerStakeMsgProcessorTest
+        ownerAddressProposition
+      )
+      val addNewStakeData: Array[Byte] = cmdInput.encode()
+
+      val stakeAmount = validWeiAmount
+      val addNewStakeMsg = getMessage(contractAddress, stakeAmount, BytesUtils.fromHexString(AddNewStakeCmd) ++ addNewStakeData, randomNonce)
+      val expStakeId = forgerStakeMessageProcessor.getStakeId(addNewStakeMsg)
+      val forgingStakeInfo = AccountForgingStakeInfo(expStakeId, ForgerStakeData(ForgerPublicKeys(blockSignerProposition, vrfPublicKey), ownerAddressProposition, stakeAmount))
+      val addNewStakeReturnData = withGas(forgerStakeMessageProcessor.process(addNewStakeMsg, view, _, defaultBlockContext))
+      assertNotNull(addNewStakeReturnData)
+
+
+      val nonce = randomNonce
+      val msgToSign = ForgerStakeMsgProcessor.getMessageToSign(forgingStakeInfo.stakeId, origin, nonce.toByteArray)
+      val msgSignatureData = Sign.signMessage(msgToSign, pair, true)
+      val msgSignature = new SignatureSecp256k1(msgSignatureData)
+
+      // create command arguments
+      val removeCmdInput = RemoveStakeCmdInput(forgingStakeInfo.stakeId, msgSignature)
+      val data: Array[Byte] = removeCmdInput.encode()
+      val msg = getMessage(contractAddress, BigInteger.valueOf(-1), BytesUtils.fromHexString(RemoveStakeCmd) ++ data, nonce)
+
+      // try processing the removal of stake, should succeed
+      val returnData = intercept[ExecutionRevertedException] {
+        withGas(forgerStakeMessageProcessor.process(msg, view, _, defaultBlockContext))
+      }
+
+      assert(returnData.getMessage === "Call value must be zero")
+
+      view.commit(bytesToVersion(getVersion.data()))
+    }
+  }
+
+  @Test
+  def testRejectSendingPositiveValueToGetAllForgerStakes(): Unit = {
+    val expectedBlockSignerProposition = "aa22334455667788112233445586778811223344556677881122334455667788" // 32 bytes
+    val blockSignerProposition = new PublicKey25519Proposition(BytesUtils.fromHexString(expectedBlockSignerProposition)) // 32 bytes
+    val expectedVrfKey = "aabbccddeeff0099aabb87ddeeff0099aabbccddeeff0099aabbccd2aeff001234"
+    val vrfPublicKey = new VrfPublicKey(BytesUtils.fromHexString(expectedVrfKey)) // 33 bytes
+
+    Mockito.when(mockNetworkParams.restrictForgers).thenReturn(true)
+    Mockito.when(mockNetworkParams.allowedForgersList).thenReturn(Seq(
+      (blockSignerProposition, vrfPublicKey)
+    ))
+
+    usingView(forgerStakeMessageProcessor) { view =>
+
+      forgerStakeMessageProcessor.init(view)
+
+      val msg = getMessage(contractAddress, BigInteger.ONE, BytesUtils.fromHexString(GetListOfForgersCmd), randomNonce)
+      withGas { gas =>
+        val returnData = intercept[ExecutionRevertedException] {
+          forgerStakeMessageProcessor.process(msg, view, gas, defaultBlockContext)
+        }
+        assert(returnData.getMessage === "Call value must be zero")
+      }
+    }
+  }
+
+  @Test
+  def testRejectSendingNegativeValueToGetAllForgerStakes(): Unit = {
```

Comment:
Same here, why duplicate the whole test? Just execute two messages, one with positive, one with negative sign.